pkg/checks/health.go

```diff
@@ -177,21 +212,22 @@ func (h *Health) check(ctx context.Context) healthData {
 
 		go func() {
 			defer wg.Done()
-
-			targetData := Target{
-				Target: target,
-				Status: "healthy",
-			}
+			state := 1
 
 			l.Debug("Starting retry routine to get health of target")
 			if err := getHealthRetry(ctx); err != nil {
-				targetData.Status = "unhealthy"
+				state = 0
 			}
 
-			l.Debug("Successfully got health status of target", "status", targetData.Status)
+			l.Debug("Successfully got health status of target", "status", stateMapping[state])
 			mu.Lock()
-			hd.Targets = append(hd.Targets, targetData)
+			hd.Targets = append(hd.Targets, Target{
+				Target: target,
+				Status: stateMapping[state],
+			})
 			mu.Unlock()
+
+			h.metrics.health.WithLabelValues(target).Set(float64(state))
```

Comment:
Same concern here as in the latency check; shouldn't this be guarded by the mutex as well?

Reply:
mutex is locked for setting the metrics now